publishToConfluence - Images not found

When running `./dtcw.bat publishToConfluence` images are not found but the page is created or updated.

**To Reproduce**
Steps to reproduce the behavior:
1. Download default template (I selected DE withhelp)
2. Configure the confluence task
3. Run `./dtcw.bat publishToConfluence` with Bearer Token
4. See error

**Expected behavior**
I would expect that all pages with there attachments (images) will be uploaded to confluence.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Configuration**

```
dtcw 0.52 - ##DTCW_GIT_HASH##
docToolchain 3.1.1
OS/arch: pwsh windows x64
Available docToolchain environments: local docker
Environments with docToolchain [3.1.1]: docker
Using environment: docker
```

```yaml
confluence.with {
    input = [
            [ file: "build/html5/arc42/arc42.html", ancestorName: 'arc42-Test' ],
    ]

    api = 'https://xyz.com/confluence/rest/api/'

    // if true API V1 only will be used. Default is true.
    // useV1Api = true

    // the key of the confluence space to write to
    spaceKey = 'XYZ'

    // variable to determine how many layers of sub pages should be created
    subpagesForSections = 1

    pagePrefix = ''

    pageSuffix = ''

    pageVersionComment = ''

    bearerToken = System.properties.confluenceApiKey

    //optional API-token to be added in case the credentials are needed for user and password exchange.
    //apikey = "[API-token]"

    extraPageContent = ''
    enableAttachments = false
    pageLimit = 20
}
```

Error message:

```
docToolchain> docDir: /project
Start using bearer auth
publish build/html5/arc42/arc42.html
Retrieved pageId for given ancestorName 'arc42-Test' is 2278643048
Template
    attachment src: https://arc42.org
.
pages retrieved
> created page 2279268763
1. Einführung und Ziele
Could not find embedded image at a known location
Embedded Image Hash efea5aa18eabc32c14a228d1313f43c7
Creating image at /project/build/html5/arc42//confluence/images/
    image: /project/build/html5/arc42/confluence/images/efea5aa18eabc32c14a228d1313f43c7.png
    attachment src: #_aufgabenstellung
    attachment src: https://docs.arc42.org/section-1/
    attachment src: #_qualitätsziele
    attachment src: #_stakeholder
allPages already retrieved
pages retrieved
> created page 2279268764

> Task :publishToConfluence FAILED

FAILURE: Build failed with an exception.

* Where:
Script '/home/dtcuser/docToolchain/scripts/publishToConfluence.gradle' line: 47

* What went wrong:
Execution failed for task ':publishToConfluence'.
> groovyx.net.http.HttpResponseException: status code: 404
```

**Additional context**
The image `build/html5/arc42/confluence/images/efea5aa18eabc32c14a228d1313f43c7.png` was generated and can be found in this folder.

--- a/changelog.adoc
+++ b/changelog.adoc
@@ -9,9 +9,13 @@ and this project tries to adhere to https://semver.org/spec/v2.0.0.html[Semantic
 
 === fixes
 
+* https://github.com/docToolchain/docToolchain/issues/1291[#1291: publishToConfluence - Images not found]
+
 === added
 
 === changed
+
+* The Confluence API configuration is now aware of contexts and hides the API version specific settings.
 
 == 3.1.1 - 2023-11-07
 

--- a/core/src/main/groovy/org/docToolchain/atlassian/clients/ConfluenceClient.groovy
+++ b/core/src/main/groovy/org/docToolchain/atlassian/clients/ConfluenceClient.groovy
@@ -5,6 +5,7 @@ import groovyx.net.http.HTTPBuilder
 import groovyx.net.http.HttpResponseException
 import groovyx.net.http.Method
 import groovyx.net.http.RESTClient
+import groovyx.net.http.URIBuilder
 import org.apache.http.entity.mime.HttpMultipartMode
 import org.apache.http.entity.mime.MultipartEntity
 import org.apache.http.entity.mime.content.InputStreamBody
@@ -13,18 +14,26 @@ import org.docToolchain.configuration.ConfigService
 
 abstract class ConfluenceClient {
 
-    protected final String API_V1_DEFAULT_PATH = "/wiki/rest/api/"
-    protected final String API_V2_DEFAULT_PATH = "/wiki/api/v2/"
-
+    private final String API_DEFAULT_CONTEXT = "wiki"
+    private final String API_V1_IDENTIFIER = "/rest/api"
+    private final String API_V2_IDENTIFIER = "/api/v2"
+    protected final String API_V1_PATH
+    protected final String API_V2_PATH
     protected final String editorVersion
+
     protected String baseApiUrl
+    private String apiContext
+
     Map headers
     RESTClient restClient
     //TODO this is a workaround for the fact that the RESTClient does not support mulitpart/form-data from Groovy 3.0.0 on
     Map proxyConfig = [:]
 
     ConfluenceClient(ConfigService configService) {
-        this.baseApiUrl = configService.getConfigProperty('confluence.api')
+        String apiConfigItem = configService.getConfigProperty('confluence.api')
+        this.baseApiUrl = buildApiBaseUrlAndSetAPIContextFromConfigItem(apiConfigItem)
+        this.API_V1_PATH = apiContext + API_V1_IDENTIFIER
+        this.API_V2_PATH = apiContext + API_V2_IDENTIFIER
         this.restClient = new RESTClient(baseApiUrl)
         restClient.setEncoderRegistry(new EncoderRegistry( charset: 'utf-8' ))
         this.headers = ['X-Atlassian-Token':'no-check']
@@ -50,6 +59,45 @@ abstract class ConfluenceClient {
         }
     }
 
+    protected String buildApiBaseUrlAndSetAPIContextFromConfigItem(String configItem) {
+        URIBuilder builder = new URIBuilder(configItem)
+        StringBuilder apiUrlBuilder = new StringBuilder(builder.getScheme()).append("://").append(builder.getHost())
+        if(builder.getPort() != -1){
+            apiUrlBuilder.append(":").append(builder.getPort())
+        }
+        String apiContext = determineApiContext(builder.getPath())
+        if(!apiContext.isEmpty()){
+            setAPIContext("/" + apiContext)
+        } else {
+            setAPIContext("")
+        }
+        return apiUrlBuilder.toString()
+    }
+
+    private setAPIContext(String apiContext) {
+        this.apiContext = apiContext
+    }
+
+    private determineApiContext(String apiPath) {
+        if(apiPath.length() == 0){
+            // no context has been set
+            return API_DEFAULT_CONTEXT
+        }
+        // remove leading slash, remove api versions identifier from path
+        apiPath = apiPath
+            .substring(1)
+            .replace(API_V1_IDENTIFIER, "")
+            .replace(API_V2_IDENTIFIER, "")
+        // then split by slash to get the context
+        String[] pathParts = apiPath.split("/")
+        if(pathParts.size() == 1){
+            // context has been set
+            return pathParts[0]
+        }
+        // assume that context has been omitted intentionally https://docs.atlassian.com/ConfluenceServer/rest/8.6.1/
+        return ""
+    }
+
     def addHeader(key, value) {
         this.headers.put(key, value)
     }
@@ -67,7 +115,6 @@ abstract class ConfluenceClient {
     abstract attachmentHasChanged(attachment, localHash)
 
     protected uploadAttachment(uri, InputStream inputStream, String fileName, note, localHash) {
-        //TODO build sane URL
         def builder = new HTTPBuilder(baseApiUrl + uri)
         //TODO this is a workaround for the fact that the RESTClient does not support mulitpart/form-data from Groovy 3.0.0 on
         if (!proxyConfig.isEmpty()) {

--- a/core/src/main/groovy/org/docToolchain/atlassian/clients/ConfluenceClientV1.groovy
+++ b/core/src/main/groovy/org/docToolchain/atlassian/clients/ConfluenceClientV1.groovy
@@ -1,47 +1,36 @@
 package org.docToolchain.atlassian.clients
 
 import groovyx.net.http.ContentType
-import groovyx.net.http.URIBuilder
 import org.docToolchain.configuration.ConfigService
 
 class ConfluenceClientV1 extends ConfluenceClient {
 
-    protected final String API_V1_PATH
 
     ConfluenceClientV1(ConfigService configService) {
         super(configService)
-        API_V1_PATH = getRealApiPath()
     }
 
     @Override
     def verifyCredentials() {
         trythis {
             restClient.get(
-                path: API_V1_PATH + 'user/current', headers: headers
+                path: API_V1_PATH + '/user/current', headers: headers
             )
         }
-    }
-
-    protected String getRealApiPath() {
-        if(this.baseApiUrl.contains("/rest/api")) {
-            String path = new URIBuilder(this.baseApiUrl).getPath()
-            return path.endsWith("/") ? path : path + "/"
-        }
-        return API_V1_DEFAULT_PATH
     }
 
     @Override
     def addLabel(pageId, label) {
         trythis {
             restClient.post(contentType: ContentType.JSON,
-                path: API_V1_PATH + 'content/' + pageId + "/label", body: label, headers: headers)
+                path: API_V1_PATH + '/content/' + pageId + "/label", body: label, headers: headers)
         }
     }
 
     @Override
     def getAttachment(Object pageId, Object fileName) {
         restClient.get(
-            path: API_V1_PATH + 'content/' + pageId + '/child/attachment',
+            path: API_V1_PATH + '/content/' + pageId + '/child/attachment',
             query: [
                 'filename': fileName,
             ], headers: headers)
@@ -49,13 +38,14 @@ class ConfluenceClientV1 extends ConfluenceClient {
 
     @Override
     def updateAttachment(String pageId, String attachmentId, InputStream inputStream, String fileName, String note, String localHash) {
-        def uri = API_V1_PATH + 'content/' + pageId + '/child/attachment/' + attachmentId + '/data'
+        def uri = API_V1_PATH + '/content/' + pageId + '/child/attachment/' + attachmentId + '/data'
         uploadAttachment(uri, inputStream, fileName, note, localHash)
     }
 
     @Override
     def createAttachment(String pageId, InputStream inputStream, String fileName, String note, String localHash) {
-        def uri = API_V1_PATH + 'content/' + pageId + '/child/attachment'
+        def uri = API_V1_PATH + '/content/' + pageId + '/child/attachment'
+        println("Uploading attachment to ${uri}")
         uploadAttachment(uri, inputStream, fileName, note, localHash)
     }
 
@@ -81,7 +71,7 @@ class ConfluenceClientV1 extends ConfluenceClient {
                 ]
                 def results = restClient.get(
                     'headers': headers,
-                    'path'   : API_V1_PATH + "content",
+                    'path'   : API_V1_PATH + "/content",
                     'query'  : query
                 ).data
                 results = results.results ?: []
@@ -122,7 +112,7 @@ class ConfluenceClientV1 extends ConfluenceClient {
             trythis {
                 def results = restClient.get(
                     'headers': headers,
-                    'path'   : API_V1_PATH + "content/${pageId}/child/page",
+                    'path'   : API_V1_PATH + "/content/${pageId}/child/page",
                     'query'  : query
                 )
                 results = results.data.results ?: []
@@ -158,7 +148,7 @@ class ConfluenceClientV1 extends ConfluenceClient {
     @Override
     def fetchPageByPageId(String id) {
         restClient.get(
-                path   : API_V1_PATH + "content/${id}",
+                path   : API_V1_PATH + "/content/${id}",
                 query: [
                         'expand': 'body.storage,version,ancestors'
                 ], headers: headers)
@@ -173,7 +163,7 @@ class ConfluenceClientV1 extends ConfluenceClient {
         restClient.get(
                 [
                         'headers': headers,
-                        'path'   : API_V1_PATH + "content",
+                        'path'   : API_V1_PATH + "/content",
                         'query'  : request,
                 ]
         ).data.results?.getAt(0)?.id
@@ -187,7 +177,7 @@ class ConfluenceClientV1 extends ConfluenceClient {
         trythis {
             restClient.put(contentType: ContentType.JSON,
                 requestContentType : ContentType.JSON,
-                path: API_V1_PATH + 'content/' + pageId, body: requestBody, headers: headers)
+                path: API_V1_PATH + '/content/' + pageId, body: requestBody, headers: headers)
         }
     }
 
@@ -198,7 +188,7 @@ class ConfluenceClientV1 extends ConfluenceClient {
         trythis {
             restClient.post(contentType: ContentType.JSON,
                 requestContentType: ContentType.JSON,
-                path: API_V1_PATH + 'content', body: requestBody, headers: headers)
+                path: API_V1_PATH + '/content', body: requestBody, headers: headers)
         }
     }
 

--- a/core/src/main/groovy/org/docToolchain/atlassian/clients/ConfluenceClientV2.groovy
+++ b/core/src/main/groovy/org/docToolchain/atlassian/clients/ConfluenceClientV2.groovy
@@ -17,14 +17,14 @@ class ConfluenceClientV2 extends ConfluenceClient {
     def verifyCredentials() {
         trythis {
             restClient.get(
-                path: API_V1_DEFAULT_PATH + 'user/current', headers: headers
+                path: API_V1_PATH + '/user/current', headers: headers
             )
         }
     }
 
     def fetchSpaceIdByKey(String spaceKey) {
             return restClient.get(
-                path   : API_V2_DEFAULT_PATH + "spaces",
+                path   : API_V2_PATH + "/spaces",
                 headers: headers,
                 query  : [
                     'keys' : [spaceKey],
@@ -38,7 +38,7 @@ class ConfluenceClientV2 extends ConfluenceClient {
     def addLabel(Object pageId, Object label) {
         trythis {
             restClient.post(contentType: ContentType.JSON,
-                path: API_V1_DEFAULT_PATH + 'content/' + pageId + "/label", body: label, headers: headers)
+                path: API_V1_PATH + '/content/' + pageId + "/label", body: label, headers: headers)
         }
     }
 
@@ -46,7 +46,7 @@ class ConfluenceClientV2 extends ConfluenceClient {
     def getAttachment(Object pageId, Object fileName) {
         //TODO NOT Found
         restClient.get(
-            path: API_V2_DEFAULT_PATH + 'pages/' + pageId + '/attachments',
+            path: API_V2_PATH + '/pages/' + pageId + '/attachments',
             query: [
                 'filename': fileName,
             ], headers: headers)
@@ -54,13 +54,13 @@ class ConfluenceClientV2 extends ConfluenceClient {
 
     @Override
     def updateAttachment(String pageId, String attachmentId, InputStream inputStream, String fileName, String note, String localHash) {
-        def uri = API_V1_DEFAULT_PATH + 'content/' + pageId + '/child/attachment/' + attachmentId + '/data'
+        def uri = API_V1_PATH + '/content/' + pageId + '/child/attachment/' + attachmentId + '/data'
         uploadAttachment(uri, inputStream, fileName, note, localHash)
     }
 
     @Override
     def createAttachment(String pageId, InputStream inputStream, String fileName, String note, String localHash) {
-        def uri = API_V1_DEFAULT_PATH + 'content/' + pageId + '/child/attachment'
+        def uri = API_V1_PATH + '/content/' + pageId + '/child/attachment'
         uploadAttachment(uri, inputStream, fileName, note, localHash)
     }
 
@@ -87,7 +87,7 @@ class ConfluenceClientV2 extends ConfluenceClient {
 
                     def response = restClient.get(
                         'headers': headers,
-                        'path'   : API_V2_DEFAULT_PATH + "spaces/${spaceId}/pages",
+                        'path'   : API_V2_PATH + "/spaces/${spaceId}/pages",
                         'query'  : query
                     ).data
                     def results = response.results ?: []
@@ -128,7 +128,7 @@ class ConfluenceClientV2 extends ConfluenceClient {
             trythis {
                 def response = restClient.get(
                     'headers': headers,
-                    'path': API_V2_DEFAULT_PATH + "pages/${pageId}/children",
+                    'path': API_V2_PATH + "/pages/${pageId}/children",
                     'query': query
                 ).data
                 def results = response.results ?: []
@@ -165,7 +165,7 @@ class ConfluenceClientV2 extends ConfluenceClient {
             'body-format'   : 'storage'
         ]
         restClient.get(
-            path   : API_V2_DEFAULT_PATH + "pages/${id}",
+            path   : API_V2_PATH + "/pages/${id}",
             headers: headers,
             query  : query
         )
@@ -180,7 +180,7 @@ class ConfluenceClientV2 extends ConfluenceClient {
         restClient.get(
             [
                 'headers': headers,
-                'path'   : API_V2_DEFAULT_PATH + "spaces/${spaceId}/pages",
+                'path'   : API_V2_PATH + "/spaces/${spaceId}/pages",
                 'query'  : request,
             ]
         ).data.results?.getAt(0)?.id
@@ -207,7 +207,7 @@ class ConfluenceClientV2 extends ConfluenceClient {
         trythis {
             restClient.put(contentType: ContentType.JSON,
                 requestContentType : ContentType.JSON,
-                path: API_V2_DEFAULT_PATH + 'pages/' + pageId, body: requestBody, headers: headers)
+                path: API_V2_PATH + '/pages/' + pageId, body: requestBody, headers: headers)
         }
     }
 
@@ -230,7 +230,7 @@ class ConfluenceClientV2 extends ConfluenceClient {
         trythis {
             restClient.post(contentType: ContentType.JSON,
                 requestContentType: ContentType.JSON,
-                path: API_V2_DEFAULT_PATH + 'pages', body: requestBody, headers: headers)
+                path: API_V2_PATH + '/pages', body: requestBody, headers: headers)
         }
     }
 }

--- a/core/src/test/groovy/org/docToolchain/atlassian/clients/ConfluenceClientSpec.groovy
+++ b/core/src/test/groovy/org/docToolchain/atlassian/clients/ConfluenceClientSpec.groovy
@@ -19,6 +19,8 @@ abstract class ConfluenceClientSpec extends Specification {
         then: "the client is created and configured with basic auth"
             confluenceClient != null
             confluenceClient.baseApiUrl == "https://confluence.atlassian.com"
+            confluenceClient.API_V1_PATH == "/wiki/rest/api"
+            confluenceClient.API_V2_PATH == "/wiki/api/v2"
             confluenceClient.headers.size() == 2
             confluenceClient.headers.get("X-Atlassian-Token") == "no-check"
             confluenceClient.headers.get("Authorization") == "Basic user:password"
@@ -38,6 +40,8 @@ abstract class ConfluenceClientSpec extends Specification {
         then: "the client is created with basic auth and api key"
             confluenceClient != null
             confluenceClient.baseApiUrl == "https://confluence.atlassian.com"
+            confluenceClient.API_V1_PATH == "/wiki/rest/api"
+            confluenceClient.API_V2_PATH == "/wiki/api/v2"
             confluenceClient.headers.size() == 3
             confluenceClient.headers.get("X-Atlassian-Token") == "no-check"
             confluenceClient.headers.get("Authorization") == "Basic user:password"
@@ -162,5 +166,86 @@ abstract class ConfluenceClientSpec extends Specification {
                        'page 5': [title: 'page 5', id: '71210367', parentId: '183954870'],
                        'page 6': [title: 'page 6', id: '76418864', parentId: '183954870'],
                        'page 7': [title: 'page 7', id: '71208921', parentId: '183954870']]
+    }
+
+    def "test ConfluencClient API path without context"() {
+        given: "i create a Confluence config and the API does not have a context path"
+            ConfigObject config = new ConfigObject()
+            config.confluence = [
+                api: "https://confluence.atlassian.com/rest/api/",
+                credentials: "user:password"
+            ]
+            ConfigService configService = new ConfigService(config)
+            ConfluenceClient confluenceClient = getConfluenceClient(configService)
+        when: "the client has been created"
+            def baseApiUrl = confluenceClient.baseApiUrl
+        then: "the API BaseUrl is set without the context path"
+            baseApiUrl == "https://confluence.atlassian.com"
+            confluenceClient.API_V1_PATH == "/rest/api"
+    }
+
+    def "test default API path is set correctly if there are no trailing slashes"() {
+        given: "i create a Confluence config with host only"
+            ConfigObject config = new ConfigObject()
+            config.confluence = [
+                api: "https://confluence.atlassian.com/wiki/api/v2",
+                credentials: "user:password"
+            ]
+            ConfigService configService = new ConfigService(config)
+            ConfluenceClient confluenceClient = getConfluenceClient(configService)
+        when: "the client has been created"
+            def baseApiUrl = confluenceClient.baseApiUrl
+        then: "the default API path is set correctly"
+            baseApiUrl == "https://confluence.atlassian.com"
+            confluenceClient.API_V1_PATH == "/wiki/rest/api"
+    }
+
+    def "test ConfluencClient API path could be context aware"() {
+        given: "i create a Confluence config that includes the API context path and has no trailing slash"
+            ConfigObject config = new ConfigObject()
+            config.confluence = [
+                api: "https://confluence.atlassian.com/foo/rest/api/",
+                credentials: "user:password"
+            ]
+            ConfigService configService = new ConfigService(config)
+            ConfluenceClient confluenceClient = getConfluenceClient(configService)
+        when: "the client has been created"
+            def baseApiUrl = confluenceClient.baseApiUrl
+        then: "the API path is set with the context path and a trailing slash"
+            baseApiUrl == "https://confluence.atlassian.com"
+            confluenceClient.API_V1_PATH == "/foo/rest/api"
+    }
+
+    def "test ConfluencClient API path could contain context only"() {
+        given: "i create a Confluence config that includes the API context only"
+            ConfigObject config = new ConfigObject()
+            config.confluence = [
+                api: "https://confluence.atlassian.com/foo",
+                credentials: "user:password"
+            ]
+            ConfigService configService = new ConfigService(config)
+            ConfluenceClient confluenceClient = getConfluenceClient(configService)
+        when: "the client has been created"
+            def baseApiUrl = confluenceClient.baseApiUrl
+        then: "the API path is set with the context path and a trailing slash"
+            baseApiUrl == "https://confluence.atlassian.com"
+            confluenceClient.API_V1_PATH == "/foo/rest/api"
+    }
+
+    def "test default API path is set correctly"() {
+        given: "i create a Confluence config with host only"
+            ConfigObject config = new ConfigObject()
+            config.confluence = [
+                api: "https://confluence.atlassian.com",
+                credentials: "user:password"
+            ]
+            ConfigService configService = new ConfigService(config)
+            ConfluenceClient confluenceClient = getConfluenceClient(configService)
+        when: "the client has been created"
+            def baseApiUrl = confluenceClient.baseApiUrl
+        then: "the default API path is set correctly"
+            baseApiUrl == "https://confluence.atlassian.com"
+            confluenceClient.API_V1_PATH == "/wiki/rest/api"
+            confluenceClient.API_V2_PATH == "/wiki/api/v2"
     }
 }

--- a/core/src/test/groovy/org/docToolchain/atlassian/clients/ConfluenceClientV1Spec.groovy
+++ b/core/src/test/groovy/org/docToolchain/atlassian/clients/ConfluenceClientV1Spec.groovy
@@ -31,51 +31,6 @@ class ConfluenceClientV1Spec extends ConfluenceClientSpec {
         return new ConfluenceClientV1(configService)
     }
 
-    def "test ConfluencClientV1 API path without context"() {
-        given: "i create a Confluence config and the API does not have a context path"
-            ConfigObject config = new ConfigObject()
-            config.confluence = [
-                api: "https://confluence.atlassian.com/rest/api/",
-                credentials: "user:password"
-            ]
-            ConfigService configService = new ConfigService(config)
-            ConfluenceClientV1 confluenceClient = getConfluenceClient(configService) as ConfluenceClientV1
-        when: "the client has been created"
-            def apiPath = confluenceClient.API_V1_PATH
-        then: "the API path is set without the context path"
-            apiPath == "/rest/api/"
-    }
-
-    def "test ConfluencClientV1 API path could be context aware"() {
-        given: "i create a Confluence config that includes the API context path and has no trailing slash"
-            ConfigObject config = new ConfigObject()
-            config.confluence = [
-                api: "https://confluence.atlassian.com/foo/rest/api",
-                credentials: "user:password"
-            ]
-            ConfigService configService = new ConfigService(config)
-            ConfluenceClientV1 confluenceClient = getConfluenceClient(configService) as ConfluenceClientV1
-        when: "the client has been created"
-            def apiPath = confluenceClient.API_V1_PATH
-        then: "the API path is set with the context path and a trailing slash"
-            apiPath == "/foo/rest/api/"
-    }
-
-    def "test default API path is set correctly"() {
-        given: "i create a Confluence config with host only"
-            ConfigObject config = new ConfigObject()
-            config.confluence = [
-                api: "https://confluence.atlassian.com",
-                credentials: "user:password"
-            ]
-            ConfigService configService = new ConfigService(config)
-            ConfluenceClientV1 confluenceClient = getConfluenceClient(configService) as ConfluenceClientV1
-        when: "the client has been created"
-            def apiPath = confluenceClient.API_V1_PATH
-        then: "the default API path is set correctly"
-            apiPath == "/wiki/rest/api/"
-    }
-
     def "test attachment diff can be detected correctly"() {
         def mockedResponse =  [results:[
             [id:"att178445677", type:"attachment", status:"current", title:"01_2_iso-25010-topics-DE.drawio.png",

--- a/src/docs/020_tutorial/070_publishToConfluence.adoc
+++ b/src/docs/020_tutorial/070_publishToConfluence.adoc
@@ -277,6 +277,9 @@ api = 'https://[yourServer]/[context]'
 ----
 In this case, the correct endpoint is `https://arc42-template.atlassian.net/wiki`.
 
+[NOTE]
+The context is optional, unless it is something other that "wiki". If you would like to enforce a URL that has no context (only valid if you specified `useV1Api = true`) for Confluence Server you should specify the full API URL, like `https://confluence.example.com/rest/api`.
+
 . In `docToolchainConfig.groovy`, add the space-key, such as `8FE`.
 ----
 spaceKey = '8FE'

